dbms/src/Flash/Coprocessor/DAGStorageInterpreter.cpp

```diff
@@ -777,7 +780,17 @@ void DAGStorageInterpreter::buildLocalStreams(DAGPipeline & pipeline, size_t max
                 throw;
             }
         }
-        pipeline.streams.insert(pipeline.streams.end(), current_pipeline.streams.begin(), current_pipeline.streams.end());
+        if (has_multiple_partitions)
+            stream_pool->addPartitionStreams(current_pipeline.streams);
+        else
+            pipeline.streams.insert(pipeline.streams.end(), current_pipeline.streams.begin(), current_pipeline.streams.end());
+    }
+    if (has_multiple_partitions)
+    {
+        for (int i = 0; i < static_cast<int>(max_streams); i++)
+        {
+            pipeline.streams.push_back(std::make_shared<MultiplexInputStream>(stream_pool, context.getDAGContext()->getMPPTaskId().toString()));
```

Comment:
```suggestion
            pipeline.streams.push_back(std::make_shared<MultiplexInputStream>(stream_pool, log->identifier()));
```

Reply:
I think we'd better decouple the log ID of underlying inpustream from that of DAGStorageInterpreter. Since it is independent of DAGStorageInterpreter. We'd better directly use mpptask ID, and prevent impact of changing DAGStorageInterpreter.

Reply:
for `batchCopRead`, mpptaskId is unkownMPPTaskId.

Reply:
from https://pingcap.feishu.cn/docs/doccnCx7gLyillJD71UtiS9ovcS, It is better that we pass `log->identifier`.

Reply:
the doc says identifier maybe changed，so we need pass root identifier

Reply:
for `batchCopRead`, MPPTaskId is meaningless.
we should `context.getDAGContext()->isMPPTask() ? context.getDAGContext()->getMPPTaskId().toString()) : log->identifier()`
https://github.com/pingcap/tiflash/blob/1dfe7fc9eba8091348314a52f495a8206ce4a02c/dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp#L1154-L1155

Reply:
But if TiFlash supports multiple table scans, we have no way to distinguish different log prefixes for different table scans.

Reply:
why？ the log has name & identifier

Reply:
> for `batchCopRead`, MPPTaskId is meaningless.
> 
> we should `context.getDAGContext()->isMPPTask() ? context.getDAGContext()->getMPPTaskId().toString()) : log->identifier()`
> 
> https://github.com/pingcap/tiflash/blob/1dfe7fc9eba8091348314a52f495a8206ce4a02c/dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp#L1154-L1155

that's fine

Reply:
such as `select * from a join b`;
the logPrefix for `MultiplexInputStream` of table a and b will both be `Multiplex: MPP<${tso}, ${task_id}>`

now the logPrefix for `ExchangeReceiver` is `ExchangeReceiver: MPP<${tso}, ${task_id}> ${executor_id}`

I think if TiFlash supports multiple table scans, the req_id of `DAGStorageInterpreter` will be `MPP<${tso}, ${task_id}> ${executor_id}`,
and the logPrefix for `MultiplexInputStream` will be `MPP<${tso}, ${task_id}> ${executor_id}`. and then we can distinguish different table scan.

Reply:
`be polluted` Exactly what we want :).
`context.getDAGContext()->isMPPTask() ? context.getDAGContext()->getMPPTaskId().toString()) : log->identifier()` is ok to me. 
We can discuss this later when we support multiple table scans.
How about adding a todo about distinguishing different table scans.

Reply:
> such as `select * from a join b`; the logPrefix for `MultiplexInputStream` of table a and b will both be `Multiplex: MPP<${tso}, ${task_id}>`
> 
> now the logPrefix for `ExchangeReceiver` is `ExchangeReceiver: MPP<${tso}, ${task_id}> ${executor_id}`
> 
> I think if TiFlash supports multiple table scans, the req_id of `DAGStorageInterpreter` will be `MPP<${tso}, ${task_id}> ${executor_id}`, and the req_id for `MultiplexInputStream` will be `MPP<${tso}, ${task_id}> ${executor_id}`. and then we can distinguish different table scan.

the DAGStorageInterpreter use context.getDAGContext()->log as log identifier, how can he distinguish that?

Reply:
> such as `select * from a join b`; the logPrefix for `MultiplexInputStream` of table a and b will both be `Multiplex: MPP<${tso}, ${task_id}>`
> 
> now the logPrefix for `ExchangeReceiver` is `ExchangeReceiver: MPP<${tso}, ${task_id}> ${executor_id}`
> 
> I think if TiFlash supports multiple table scans, the req_id of `DAGStorageInterpreter` will be `MPP<${tso}, ${task_id}> ${executor_id}`, and the req_id for `MultiplexInputStream` will be `MPP<${tso}, ${task_id}> ${executor_id}`. and then we can distinguish different table scan.

I think we should concat string with root identifier and table name, not directly inherit from that of DAGStorageInterpreter.

Reply:
https://github.com/pingcap/tiflash/blob/master/dbms/src/Flash/Coprocessor/DAGStorageInterpreter.cpp#L224
change to `log(Logger::get("DAGStorageInterpreter", context.getDAGContext()->log ? context.getDAGContext()->log->identifier() : "", table_scan.getExecutorId()))`.

Reply:
> https://github.com/pingcap/tiflash/blob/master/dbms/src/Flash/Coprocessor/DAGStorageInterpreter.cpp#L224 change to `log(Logger::get("DAGStorageInterpreter", context.getDAGContext()->log ? context.getDAGContext()->log->identifier() : "", table_scan.getExecutorId()))`.

we'd better solve that thing into another indepent PR. So that we won't mess up.

Reply:
The DMSegmentInputStream use MppTaskId as identifier, so IMO this PR need to keep the same behaviour  with that. ref: [https://github.com/pingcap/tiflash/blob/master/dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp#L1208](https://github.com/pingcap/tiflash/blob/master/dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp#L1208)

Reply:
ok

Reply:
`IsMppTask` check added.